java-client/src/main/java/org/opensearch/client/opensearch/indices/IndexSettings.java

```diff
@@ -992,15 +1001,20 @@ protected void serializeInternal(JsonGenerator generator, JsonpMapper mapper) {
 			generator.write(this.maxSlicesPerScroll);
 
 		}
-		if (this.translogDurability != null) {
-			generator.writeKey("translog.durability");
-			generator.write(this.translogDurability);
+		if (this.translog != null) {
+			generator.writeKey("translog");
+			this.translog.serialize(generator, mapper);
 
-		}
-		if (this.translogFlushThresholdSize != null) {
-			generator.writeKey("translog.flush_threshold_size");
-			generator.write(this.translogFlushThresholdSize);
+			if (this.translog.durability() != null) {
```

Comment:
Sorry why do you need to serialize translog twice? 

```
this.translog.serialize(generator, mapper);
```

should have all fields already included

Reply:
OK, we could clearly drop the old `translog.xxx` fields, this is the request send to server and it will properly decode the object into the right index settings, we don't need to send it twice

Reply:
I agree, we could drop the old `translog.xxx` fields if the serialized JSON is only used inside `opensearch-java:client`, i.e. in a request to server. 
But I am not sure about the client code. It could use some custom parser to deserialize `IndexSettings` JSON into some user-provided custom structure, and it could rely on the fact that translog settings are represented as `translog.xxx` fields. Therefore, getting rid of them in the serialized JSON would break such client code.
I seems unlikely that someone would write custom deserializer, given that each class has a built-in deserializer, and it would probably be easier to use the deserialized class for all the custom operations.

Nevertheless, I can't say with confidence that there is no such "someone", so just being extra careful :)

Reply:
> But I am not sure about the client code. 

But this is the client code we are changing :) AFAIK since this is the request, out only, we never deserialize it. The only target of this request is OpenSearch server.

Reply:
OK I think I am reassured now, I am removing the redundant serialization.